TStackTest/UnitTest1.cs

```diff
@@ -0,0 +1,50 @@
+using Training;
+namespace TStackTest {
+   [TestClass]
+   public class UnitTest1 {
+      [TestMethod]
+      public void PushTest () {
+         TStack<int> numbers = new ();

```

Comment:
Instead of declaring a TStack object for each test, a private variable can be used to reduce GC (garbage collection) pressure. The stack can be cleared after each test.